atomicapp/providers/lib/kubeshift/kubernetes.py

```diff
@@ -0,0 +1,171 @@
+"""
+ Copyright 2014-2016 Red Hat, Inc.
+
+ This file is part of Atomic App.
+
+ Atomic App is free software: you can redistribute it and/or modify
+ it under the terms of the GNU Lesser General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ Atomic App is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU Lesser General Public License for more details.
+
+ You should have received a copy of the GNU Lesser General Public License
+ along with Atomic App. If not, see <http://www.gnu.org/licenses/>.
+"""
+
+from urlparse import urljoin
+from urllib import urlencode
+from atomicapp.providers.lib.kubeshift.kubebase import KubeBase
+from atomicapp.providers.lib.kubeshift.exceptions import (KubeKubernetesError)
+from atomicapp.constants import LOGGER_DEFAULT
+import logging
+import re
+logger = logging.getLogger(LOGGER_DEFAULT)
+
+
+class KubeKubernetesClient(object):
+
+    def __init__(self, config):
+        '''
+
+        Args:
+            config (obj): Object of the configuration data
+
+        '''
+
+        # Pass in the configuration data (.kube/config object) to the KubeBase
+        self.api = KubeBase(config)
+
+        # Check the API url
+        url = self.api.cluster['server']
+        if not re.match('(?:http|https)://', url):
+            raise KubeKubernetesError("Kubernetes API URL does not include HTTP or HTTPS")
+
+        # Gather what end-points we will be using
+        self.k8s_api = urljoin(url, "api/v1/")
```

Comment:
@khrisrichardson We hope so. Any changes to https://github.com/projectatomic/nulecule will reflect in atomic app in terms of providing support for autoscaling, batch commands, extensions, policies, etc. If the spec changes we'll reflect that as features in Atomic App :)